Makefile

```diff
@@ -23,29 +23,36 @@ links:
 	$(LN) ../../src/rand/rand.h include/oqs
 	$(LN) ../../src/rand_urandom_chacha20/rand_urandom_chacha20.h include/oqs
 
-rand_urandom_chacha20: src/rand_urandom_chacha20/rand_urandom_chacha20.h src/rand_urandom_chacha20/rand_urandom_chacha20.c src/rand_urandom_chacha20/chacha20.c
-	$(CC) -c $(CFLAGS) $(INCLUDES) src/rand_urandom_chacha20/rand_urandom_chacha20.c -o objs/rand_urandom_chacha20.o
-	$(AR) liboqs.a objs/rand_urandom_chacha20.o
-
-rand: src/rand/rand.h src/rand/rand.c
-	$(CC) -c $(CFLAGS) $(INCLUDES) src/rand/rand.c -o objs/rand.o
-	$(AR) liboqs.a objs/rand.o
-
-kex_rlwe_bcns15: src/kex_rlwe_bcns15/fft.c src/kex_rlwe_bcns15/kex_rlwe_bcns15.c src/kex_rlwe_bcns15/kex_rlwe_bcns15.h src/kex_rlwe_bcns15/local.h src/kex_rlwe_bcns15/rlwe.c src/kex_rlwe_bcns15/rlwe_a.h src/kex_rlwe_bcns15/rlwe_kex.c src/kex_rlwe_bcns15/rlwe_table.h
-	$(CC) -c $(CFLAGS) $(INCLUDES) src/kex_rlwe_bcns15/fft.c -o objs/kex_rlwe_bcns15_fft.o
-	$(CC) -c $(CFLAGS) $(INCLUDES) src/kex_rlwe_bcns15/rlwe.c -o objs/kex_rlwe_bcns15_rlwe.o
-	$(CC) -c $(CFLAGS) $(INCLUDES) src/kex_rlwe_bcns15/rlwe_kex.c -o objs/kex_rlwe_bcns15_rlwe_kex.o
-	$(CC) -c $(CFLAGS) $(INCLUDES) src/kex_rlwe_bcns15/kex_rlwe_bcns15.c -o objs/kex_rlwe_bcns15.o
-	$(AR) liboqs.a objs/kex_rlwe_bcns15_fft.o
-	$(AR) liboqs.a objs/kex_rlwe_bcns15_rlwe.o
-	$(AR) liboqs.a objs/kex_rlwe_bcns15_rlwe_kex.o
-	$(AR) liboqs.a objs/kex_rlwe_bcns15.o
-
-kex: src/kex/kex.h src/kex/kex.c
-	$(CC) -c $(CFLAGS) $(INCLUDES) src/kex/kex.c -o objs/kex.o
-	$(AR) liboqs.a objs/kex.o
-
-lib:
+# CHACHA
+
+CHACHA_OBJS :=  $(addprefix objs/rand_urandom_chacha20/, \
+	            rand_urandom_chacha20.o chacha20.o)
+
+$(CHACHAOBJS): src/rand_urandom_chacha20/rand_urandom_chacha20.h
```

Comment:
I'm not too familiar with Makefile syntax.  Do you intentionally have CHACHA_OBJS in one spot and CHACHAOBJS in another?